libs/chonky/src/main.rs

```diff
@@ -0,0 +1,36 @@
+use std::env;
+
+use chonky::{ChonkyError, load_pdf, pdf_to_images};
+use error_stack::{Report, ResultExt};
+use pdfium_render::prelude::*;
+
+fn main() -> Result<(), Report<ChonkyError>> {
+    let args: Vec<String> = env::args().collect(); // read file path arguments
+
+    if args.len() < 2 {
+        return Err(Report::new(ChonkyError::Arguments));
+    }
```

Comment:
`error_stack` has a handy [macro](https://docs.rs/error-stack/latest/error_stack/macro.ensure.html) for this:
```rust
use error_stack::ensure;
```

```suggestion
    ensure!(args.len() > 1, ChonkyError::Arguments);
```

But this is only really usable for simple errors such as this one. The macro is agnostic over the second argument, so it can be either a `Report` or a plain error, the result will always be a `Report` wrapped in an `Err`.

Reply:
cc @indietyp the documentation seems a bit broken, the `stringify` is not supposed to be shown I guess 😄 
<img width="675" alt="image" src="https://github.com/user-attachments/assets/31109b96-e256-4b76-82db-028dd3de1e49">
I'm going to look into that at some point later: H-3536

Reply:
^ Fixed as part of #5533